mssql-plugin/src/e2e-test/java/io.cdap.plugin/mssql/BQValidation.java

```diff
@@ -0,0 +1,238 @@
+        List<Object> bigQueryRows = new ArrayList<>();
+        getBigQueryTableData(sourceTable, bigQueryRows);
+        for (Object rows : bigQueryRows) {
+            JsonObject json = new Gson().fromJson(String.valueOf(rows), JsonObject.class);
+            jsonResponse.add(json);
+        }
+        String getTargetQuery = "SELECT * FROM " + schema + "." + targetTable;
+        try (Connection connect = MssqlClient.getMssqlConnection()) {
+            connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+            Statement statement1 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                    ResultSet.HOLD_CURSORS_OVER_COMMIT);
+
+            ResultSet rsTarget = statement1.executeQuery(getTargetQuery);
+            return compareResultSetAndJsonData(rsTarget, jsonResponse);
+        }
+    }
+
+    /**
+     * Retrieves the data from a specified BigQuery table and populates it into the provided list of objects.
+     *
+     * @param table        The name of the BigQuery table to fetch data from.
+     * @param bigQueryRows The list to store the fetched BigQuery data.
+     */
+
+    private static void getBigQueryTableData(String table, List<Object> bigQueryRows)
+            throws IOException, InterruptedException {
+
+        String projectId = PluginPropertyUtils.pluginProp("projectId");
+        String dataset = PluginPropertyUtils.pluginProp("dataset");
+        String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+        TableResult result = BigQueryClient.getQueryResult(selectQuery);
+        result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+    }
+
+    /**
+     * Compares the data in the result set obtained from the Oracle database with the provided BigQuery JSON objects.
+     *
+     * @param rsSource     The result set obtained from the Oracle database.
+     * @param bigQueryData The list of BigQuery JSON objects to compare with the result set data.
+     * @return True if the result set data matches the BigQuery data, false otherwise.
+     * @throws SQLException   If an SQL error occurs during the result set operations.
+     * @throws ParseException If an error occurs while parsing the data.
+     */
+
+    public static boolean compareResultSetAndJsonData(ResultSet rsSource, List<JsonObject> bigQueryData)
+            throws SQLException, ParseException {
+        ResultSetMetaData mdSource = rsSource.getMetaData();
+        boolean result = false;
+        int columnCountSource = mdSource.getColumnCount();
+
+        if (bigQueryData == null) {
+            Assert.fail("bigQueryData is null");
+            return result;
+        }
+
+        // Get the column count of the first JsonObject in bigQueryData
+        int columnCountTarget = 0;
+        if (bigQueryData.size() > 0) {
+            columnCountTarget = bigQueryData.get(0).entrySet().size();
```

Comment:
You are simply checking the size of the columns of first row what if there are more than one rows in the big Query table. Don't hardcode it.

Reply:
Done